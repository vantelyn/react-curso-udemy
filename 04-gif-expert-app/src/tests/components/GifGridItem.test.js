import React from "react";
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en GifGridItem', () => { 

    const img = {
        id:'l3vRlt8kty8KKeHni',
        title:'one punch GIF by Adult Swim',
        url:'https://media1.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eezgvx8hg8k0cllljly13foyb81et5rgk4qnz86jex&rid=giphy.gif&ct=g',
    };

    const wrapper = shallow ( <GifGridItem { ...img } /> );

	test('Debe de mostrar el componente correctamente', () => { 
		
		expect( wrapper ).toMatchSnapshot();
								
	});

    test('debe de tener un pÃ¡rrafo con el title', () => { 

        const parrafo = wrapper.find('p').text().trim();
        expect(parrafo).toBe(img.title);

    });

    test('debe mostrar la imagen correspondiente', () => { 

        // const src = wrapper.find('img').prop('src');
        // const alt = wrapper.find('img').prop('alt');
        // const props = wrapper.find('img').props();
        const imgTest = wrapper.find('img');

        // expect(props.src).toBe(img.url);
        // expect(props.alt).toBe(img.title);
        
        expect( imgTest.prop('src') ).toBe(img.url);
        expect( imgTest.prop('alt') ).toBe(img.title);

    });

    test('debe de tener animate__fadeIn', () => { 

        const div = wrapper.find('div');
        const className = div.prop( 'className' );

        // expect( className ).toContain( 'animate__fadeIn' );
        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );
    })

}); 