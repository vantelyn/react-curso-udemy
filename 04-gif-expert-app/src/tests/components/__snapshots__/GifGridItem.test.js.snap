// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="one punch GIF by Adult Swim"
    src="https://media1.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=5bcd87eezgvx8hg8k0cllljly13foyb81et5rgk4qnz86jex&rid=giphy.gif&ct=g"
  />
  <p>
     
    one punch GIF by Adult Swim
     
  </p>
</div>
`;
